sql/binlog.h

```diff
@@ -387,6 +387,7 @@ class MYSQL_BIN_LOG : public TC_LOG {
   mysql_cond_t update_cond;
 
   std::atomic<my_off_t> atomic_binlog_end_pos;
+  std::pair<std::string, my_off_t> last_acked;
```

Comment:
typedef/using for a custom type

Reply:
Why? It's only place when this type is used.

Reply:
For me, personally, this is just a rule of thumb: if you need a complex type, declare a typedef/using for it - pretty big chances you will have to refer to it more than once. In this particular case it does not seem very important.
Leaving this up to you.